android/java/org/chromium/chrome/browser/crypto_wallet/util/ParsedTransaction.java

```diff
@@ -164,57 +133,57 @@ public static ParsedTransaction parseTransaction(
                         || txInfo.txType
                                 == TransactionType
                                         .SOLANA_SPL_TOKEN_TRANSFER_WITH_ASSOCIATED_TOKEN_ACCOUNT_CREATION;
         final boolean isSolTransaction = SOLANA_TRANSACTION_TYPES.contains(txInfo.txType);
-        final boolean isBtcTransaction = btcTxData != null;
 
-        final String value =
-                isSPLTransaction
-                        ? solTxData != null ? String.valueOf(solTxData.amount) : ""
-                        : isSolTransaction
-                                ? solTxData != null ? String.valueOf(solTxData.lamports) : ""
-                                : isFilTransaction
-                                        ? filTxData.value != null
-                                                ? Utils.toHex(filTxData.value)
-                                                : ""
-                                        : isBtcTransaction
-                                                ? String.valueOf(btcTxData.amount)
-                                                : txData != null ? txData.baseData.value : "";
-
-        String to =
-                isSolTransaction
-                        ? solTxData != null ? solTxData.toWalletAddress : ""
-                        : isFilTransaction
-                                ? filTxData.to
-                                : isBtcTransaction ? btcTxData.to : txData.baseData.to;
+        String value = "";
+        if (isSPLTransaction && solTxData != null) {
+            value = String.valueOf(solTxData.amount);
+        } else if (isSolTransaction && solTxData != null) {
+            value = String.valueOf(solTxData.lamports);
+        } else if (filTxData != null && !TextUtils.isEmpty(filTxData.value)) {
+            value = Utils.toHex(filTxData.value);
+        } else if (btcTxData != null) {
+            value = String.valueOf(btcTxData.amount);
+        } else if (zecTxData != null) {
+            value = String.valueOf(zecTxData.amount);
+        } else if (txData != null) {
+            value = txData.baseData.value;
+        }
+
+        String to = "";
+        if (isSolTransaction && solTxData != null) {
+            to = solTxData.toWalletAddress;
+        } else if (filTxData != null) {
+            to = filTxData.to;
+        } else if (btcTxData != null) {
+            to = btcTxData.to;
+        } else if (zecTxData != null) {
+            to = zecTxData.to;
+        } else if (txData != null) {
+            to = txData.baseData.to;
+        }
 
         final String nonce = txData != null ? txData.baseData.nonce : "";
         AccountInfo account = Utils.findAccount(accounts, txInfo.fromAccountId);
-
-        final String accountAddressLower =
-                account != null ? account.address.toLowerCase(Locale.getDefault()) : "";
+        if (account == null) {
+            throw new IllegalStateException("Account not found from transaction info account ID.");
```

Comment:
do we catch it in all cases? I don't see any catch on that level for example https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/crypto_wallet/fragments/ApproveTxBottomSheetDialogFragment.java#L407

Reply:
I'm going to switch to a less aggressive approach by using an assert instead of throwing, so the app won't crash. 
A `null` account should never happen in any way as it would imply that we are in an inconsistant state with an unknown sender.

Reply:
Let me know what you think about this solution 👉 [`4a51740` (#28883)](https://github.com/brave/brave-core/pull/28883/commits/4a51740df303ef53ce6cc7e19c757e3378c78914)